src/OrchardCore/OrchardCore.DisplayManagement/Zones/ZoneShapes.cs

```diff
@@ -13,43 +14,70 @@ namespace OrchardCore.DisplayManagement.Zones
     [Feature(Application.DefaultFeatureId)]
     public class ZoneShapes : IShapeAttributeProvider
     {
+        // By convention all placement delimiters default to the name 'Content' when not specified during placement.
+        private const string ContentKey = "Content";
+
         [Shape]
-        public async Task<IHtmlContent> Zone(dynamic DisplayAsync, dynamic Shape)
+        public async Task<IHtmlContent> Zone(IDisplayHelper DisplayAsync, IEnumerable<object> Shape)
         {
             var htmlContentBuilder = new HtmlContentBuilder();
             foreach (var item in Shape)
             {
-                htmlContentBuilder.AppendHtml(await DisplayAsync(item));
+                htmlContentBuilder.AppendHtml(await DisplayAsync.ShapeExecuteAsync(item));
```

Comment:
Here we can inject and use `IDisplayHelper` instead of treating it as dynamic, same with the `Shape`

No a required change, but in my mind, makes it easier to actually see what you're working with, plus less dynamic accessing